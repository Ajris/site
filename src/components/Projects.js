import React, {Component} from 'react';

export default class Projects extends Component {
    render() {
        return (
            <div className={'mx-4 my-3'}>
                <p className={'topic text-center'}>PROJECTS INFORMATION</p>
            </div>
        )
    }
}