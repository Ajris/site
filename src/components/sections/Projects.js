import React from 'react';
import CardDeck from "react-bootstrap/CardDeck";
import ItemCard from "./ItemCard";


export default class Projects extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            cards: [],
        };
    }

    componentDidMount() {
        fetch("http://localhost:8080/project")
            .then(res => res.json())
            .then((result) => {
                let i = 0;
                let cards = result.map((item) =>
                    <ItemCard key={i++} title={item.name} text={item.description} image={require("../../assets/projects/site.png")} url={item.url}/>
                );
                this.setState({cards:cards});
            });

    }

    render() {
        return (
            <div>
                <div className={'mx-4 my-3'}>
                    <p className={'topic text-center'}>Projects</p>
                </div>
                <CardDeck>
                    {this.state.cards}
                </CardDeck>
            </div>
        )
    }
}