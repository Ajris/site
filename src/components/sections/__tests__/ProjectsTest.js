import React from "react";
import { create } from "react-test-renderer";
import Projects from "../Projects";

describe("Projects component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Projects/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
