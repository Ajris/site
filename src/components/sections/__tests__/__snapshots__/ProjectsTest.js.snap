// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component it matches the snapshot 1`] = `
<div>
  <div
    className="mx-4 my-3"
  >
    <p
      className="topic text-center"
    >
      Projects
    </p>
  </div>
  <div
    className="card-deck"
  />
</div>
`;
